### PR TITLE
Add metrics to event

### PR DESCRIPTION
**Pull request by aneeshkp**
Added mock plugin and fixed metrics
Added local docker file to build local with local deps

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -236,14 +250,31 @@ func ProcessInChannel(wg *sync.WaitGroup, scConfig *common.SCConfiguration) {
...
+					if err := d.OnReceiveOverrideFn(*d.Data, &out); err != nil {
 						log.Errorf("error onReceiveOverrideFn %s", err)
 						out.Status = channel.FAILED
 					} else {
 						out.Status = channel.SUCCESS
 					}
 				}
 				scConfig.EventOutCh <- &out
+			} else if d.Type == channel.STATUS && d.Status == channel.NEW {
+				log.Warnf("amqp disabled,no action taken(can't send to a desitination): logging new status checl %v\n", d)
```
Comment (jzding) [inline cmd/main.go:262]: typo: `desitination`): logging new status `checl ` -> `destination`    `check`

File: examples/consumer/deployment/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                  - key: app
+                    operator: In
+                    values:
+                      - local
+      serviceAccountName: sidecar-consumer-sa
+      containers:
+        - name: cloud-native-event-consumer
+          image: quay.io/aneeshkp/cloud-native-event-consumer
+          args:
+            - "--local-api-addr=127.0.0.1:9089"
```
Comment (jzding) [inline examples/consumer/deployment/deployment.yaml:32]: 8989?

File: examples/consumer/main.go [hunk 3]
```diff
@@ -33,49 +33,79 @@ import (
...
+	flag.StringVar(&localAPIAddr, "local-api-addr", "localhost:8989", "The address the local api binds to .")
 	flag.StringVar(&apiPath, "api-path", "/api/cloudNotifications/v1/", "The rest api path.")
 	flag.StringVar(&apiAddr, "api-addr", "localhost:8080", "The address the framework api endpoint binds to.")
 	flag.Parse()
 
-	if envNode := os.Getenv("NODE_NAME"); envNode != "" {
-		nodeName = envNode
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		log.Error("cannot find NODE_NAME environment variable,setting to default `moc` noe")
```
Comment (jzding) [inline examples/consumer/main.go:68]: typo: `mock` node

File: examples/consumer/main.go [hunk 4]
```diff
@@ -33,49 +33,79 @@ import (
...
+		Resource: resourceAddressHwEvent,
+	}, &pubsub.PubSub{
+		Resource: resourceAddressMock,
+	}, &pubsub.PubSub{
+		Resource: fmt.Sprintf(resourceAddressPTP, nodeName),
+	},
+	}
+
+	if consumerType==PTP {
+		createPublisherForStatusPing(subs[2].Resource) // ptp // disable this for testing else you will see context deadline error
```
Comment (jzding) [inline examples/consumer/main.go:104]: any link to the issue?

File: examples/consumer/deployment/deployment.yaml (hunk 2 above)
Comment (aneeshkp) [reply examples/consumer/deployment/deployment.yaml:32]: doesn't matter

File: examples/consumer/main.go [hunk 5]
```diff
@@ -86,13 +116,88 @@ RETRY:
 		log.Infof("created subscription: %s\n", sub.String())
 	}
 
+	// ping  for status every five secs
```
Comment (jzding) [inline examples/consumer/main.go:119]: looks like it is every 30 seconds?

File: examples/consumer/main.go [hunk 6]
```diff
@@ -86,13 +116,88 @@ RETRY:
...
+		Host: localAPIAddr,
+		Path: fmt.Sprintf("%s", "event")}}
+
+	pub := v1pubsub.NewPubSub(endpointURL, resourceAddress)
+	if b, err := json.Marshal(&pub); err == nil {
+		rc := restclient.New()
+		if status, b := rc.PostWithReturn(subURL, b); status == http.StatusCreated {
+			return b
+		}
+		log.Errorf("subscription create returned error %s", string(b))
```
Comment (jzding) [inline examples/consumer/main.go:151]: it will be helpful to log status as well

File: examples/consumer/main.go [hunk 7]
```diff
@@ -86,13 +116,88 @@ RETRY:
...
+		Path: fmt.Sprintf("%s", "ack/event")}}
+	log.Infof("publisher endpoint %s", endpointURL.String())
+	pub := v1pubsub.NewPubSub(endpointURL, fmt.Sprintf("%s/%s", resourceAddress, "status"))
+	if b, err := json.Marshal(&pub); err == nil {
+		rc := restclient.New()
+		if status, b := rc.PostWithReturn(publisherURL, b); status == http.StatusCreated {
+			log.Infof("create status ping publisher %s", string(b))
+			return b
+		}
+		log.Errorf("status ping publisher create returned error %s", string(b))
```
Comment (jzding) [inline examples/consumer/main.go:175]: log status

File: examples/consumer/main.go [hunk 8]
```diff
@@ -110,23 +215,16 @@ func getEvent(w http.ResponseWriter, req *http.Request) {
...
-		log.Errorf("subscription create returned error %s", string(b))
+func ackEvent(w http.ResponseWriter, req *http.Request) {
+	defer req.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		log.Errorf("error reading acknowledgment  %v", err)
+	}
+	e := string(bodyBytes)
+	if e != "" {
+		log.Debugf("received ack %s", string(bodyBytes))
```
Comment (jzding) [inline examples/consumer/main.go:226]: use `e` instead of `string(bodyBytes)`

File: examples/producer/main.go [hunk 9]
```diff
@@ -113,12 +116,12 @@ func createPublisher(resourceAddress string) []byte {
...
+	log.Infof("publisher endpoint %s", endpointURL.String())
 	pub := v1pubsub.NewPubSub(endpointURL, resourceAddress)
 	if b, err := json.Marshal(&pub); err == nil {
 		rc := restclient.New()
 		if status, b := rc.PostWithReturn(publisherURL, b); status == http.StatusCreated {
-			log.Infof("sdss%s", string(b))
+			log.Infof("create publisher status %s", string(b))
 			return b
 		}
 		log.Errorf("publisher create returned error %s", string(b))
```
Comment (jzding) [inline examples/producer/main.go:127]: log `status` code

File: cmd/main.go [hunk 10]
```diff
@@ -68,8 +68,8 @@ func main() {
 
 	// Register metrics
 	localmetrics.RegisterMetrics()
```
Comment (jzding) [inline cmd/main.go:70]: localMetrics

File: pkg/localmetrics/localmetrics.go [hunk 11]
```diff
@@ -41,12 +41,19 @@ var (
 			Name: "cne_events_received",
 			Help: "Metric to get number of events received",
 		}, []string{"type", "status"})
+	//CNEStatusCheckReceivedCount ...  Total no of events received
+	cneStatusCheckReceivedCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "cne_status_check_received",
```
Comment (jzding) [inline pkg/localmetrics/localmetrics.go:47]: I don't see this documented in docs/metrics.md

**Pull request merged**